### DISCUSSION TITLE
fs does not write buffer

### DISCUSSION THREAD

**Discussion by burki505** [nodejs/help#4802] [Q&A]
I want to write buffered data to my file using fs but it does not work. What could be the problem?

`async import(dto: ImportDto, veritabaniId: number): Promise<Buffer> {
    const quadDto = new QuadBulkCreateDto();
    quadDto.quads = JSON.parse(dto.file.data);

    try {
      await validate(quadDto);
      const dirPath = join(envConfig.quadsDataPath, veritabaniId.toString());
      const filePath = join(dirPath, 'quad_data.json');

      await fs.promises.mkdir(dirPath, { recursive: true });

      const binaryFile = Buffer.from(JSON.stringify(quadDto.quads));

      await fs.promises.writeFile(filePath, binaryFile);

      const fileData = await fs.promises.readFile(filePath);

      return fileData;
    } catch (error) {
      throw error;
    }
  }`

**Comment by JimmyDaddy** [comment]
There is no problem about the write buffer code `await fs.promises.writeFile(filePath, binaryFile);`

```ts
import fs from 'fs';

const binaryFile = Buffer.from(JSON.stringify({
  a: 'test'
}));

fs.promises.writeFile('./test.txt', binaryFile);
// check the file test.txt content, you will get string '{"a":"test"}'

```

maybe the problem is `quadDto.quads` is an empty object/string?
